volatility3/framework/layers/resources.py

```diff
@@ -100,6 +101,21 @@ def open(self, url: str, mode: str = "rb") -> Any:
         """
         urllib.request.install_opener(urllib.request.build_opener(*self._handlers))
 
+        # Python bug 46654
+        if sys.platform == 'win32':
```

Comment:
I would move the login in this if condition to a different function that should handle quircks

Reply:
Not entirely sure what you mean?  The logging is relevant only to that area, the code block stands alone and we don't really have a section for quirks?

Reply:
I just meant to have this logic in a separate function like `get_fixed_url()`. That function will log appropriately, I just don't like having this logic inline within the `open` function

Reply:
Ok, I see.  I think it's in keeping with the rest of the codebase and I'm less keen on having bits of logic splayed out over lots of functions.  I think it's as coding style call so I'll probably leave it as is, but thanks for the feedback, I'm glad someone took a look at the code at least!  5:)